src/modules/pua_json/pua_json_publish.c

```diff
@@ -359,6 +360,57 @@ int pua_json_publish_dialoginfo_to_presentity(struct json_object *json_obj)
 	return ret;
 }
 
+int pua_json_publish_as_feature_to_presentity(struct json_object *json_obj)
+{
+	int ret = 1;
+	int len;
+	str event = str_init("as-feature-event");
+	str from = {0, 0};
+	str from_user = {0, 0};
+	str from_realm = {0, 0};
+	str callid = {0, 0};
+	str dnd_status = {0, 0};
+	int expires = 0;
+	str as_feature_body = {0, 0};
+
+	char *body = (char *)pkg_malloc(AS_FEATURE_BODY_BUFFER_SIZE);
+
+	if(body == NULL) {
+		LM_ERR("Error allocating buffer for publish\n");
+		ret = -1;
+		goto error;
+	}
+
+	json_api.extract_field(json_obj, AS_FEATURE_JSON_CALLID, &callid);
+
+	json_api.extract_field(json_obj, AS_FEATURE_JSON_FROM, &from);
+	json_api.extract_field(json_obj, AS_FEATURE_JSON_FROM_USER, &from_user);
+	json_api.extract_field(json_obj, AS_FEATURE_JSON_FROM_REALM, &from_realm);
+
+	json_api.extract_field(json_obj, AS_FEATURE_JSON_DND_STATUS, &dnd_status);
+
+	struct json_object *ExpiresObj =
+			json_api.get_object(json_obj, AS_FEATURE_JSON_EXPIRES);
+	if(ExpiresObj != NULL) {
+		expires = json_object_get_int(ExpiresObj);
+	}
+
+	len = snprintf(body, AS_FEATURE_BODY_BUFFER_SIZE, AS_FEATURE_BODY,
```

Comment:
We probably want to check if you've got a from and dnd value from the json before doing the snprintf. Compare to line 294 in this file, for example. If there are other mandatory fields we need to get from the json, we should also check for them. If they are not present, just go to the error case.

Reply:
I've added check on `from_user` and `dnd_status`